File: app/src/main/java/closer/vlllage/com/closer/store/StoreRefHandler.java
package closer.vlllage.com.closer.store;

import closer.vlllage.com.closer.handler.ApplicationHandler;
import closer.vlllage.com.closer.pool.PoolMember;

public class StoreRefHandler extends PoolMember {

    private Store store;

    @Override
    public void onPoolInit() {
        this.store = new Store($(ApplicationHandler.class).getApp());
    }

    @Override
    protected void onPoolEnd() {
        store.close();
    }

    public Store get() {
        return store;
    }
}

File: app/src/main/java/closer/vlllage/com/closer/store/StoreRefHandler.kt
package closer.vlllage.com.closer.store

import closer.vlllage.com.closer.handler.helpers.ApplicationHandler
import com.queatz.on.On
import com.queatz.on.OnLifecycle

class StoreRefHandler constructor(private val on: On) : OnLifecycle {

    private lateinit var store: Store

    override fun on() {
        this.store = Store(on<ApplicationHandler>().app)
    }

    override fun off() {
        store.close()
    }

    fun get() = store
}
